### ISSUE TITLE
Examples execution errors on OSX

### ISSUE THREAD

**Issue by Kerosene2000** [sebcrozet/nphysics#57]
It seems that `sfml/csml` cannot create the proper context for OpenGl or something like that.
When I `./examples2d/target/release/gravity --help`:

```
Warning. OpenGL debugging not supported on this platform.
Error. Unable to find a suitable pixel format.
Warning: The created OpenGL context does not fully meet the settings that were requested
Requested: version = 0.1 ; depth bits = 10 ; stencil bits = 10 ; AA level = 2 ; core = true ; debug = true
Created: version = 2.1 ; depth bits = 10 ; stencil bits = 10 ; AA level = 2 ; core = false ; debug = false
Failed to activate the window's context
Failed to activate the window's context
```

I have set the dynamics and statics libraries paths like this:
``` bash
export LD_LIBRARY_PATH=$HOME/.brew/lib/
export DYLD_LIBRARY_PATH=$HOME/.brew/lib/
```

When I execute the gravity example, sometimes I get a *segfault*, sometimes an *infinite loop* of `Failed to activate the window's context`...

**Comment by sebcrozet** [comment]
I am not sure how to identify the cause of the problem. This is probably independent from `nphysics`. Maybe you could try to run the simpler examples from [rust-sfml](https://github.com/jeremyletang/rust-sfml) and see if you get the same behavior. Unfortunately I don't have OSX so I can't reproduce it.

Maybe @jeremyletang has some ideas ?

**Comment by Kerosene2000** [comment]
I just clone the **last** version of [rust-sfml](https://github.com/jeremyletang/rust-sfml) and all examples works fine...

I will try with the version you use in a few minutes

**Comment by Kerosene2000** [comment]
I just try with the `0.9.3` and all examples works fine too...

**Comment by sebcrozet** [comment]
Strange... could you try to:

1. Replace `&setting` on [that line](https://github.com/sebcrozet/nphysics/blob/master/examples2d/nphysics_testbed2d/src/testbed.rs#L97) with `&ContextSettings::default()` and see if that helps ?
2. Run the 3D examples of `nphysics` to be sure this is only a sfml-related problem.

**Comment by Kerosene2000** [comment]
All examples works greats with this modification...
![gravity](https://cloud.githubusercontent.com/assets/3610253/14281837/b6f46ab2-fb3b-11e5-8d40-a09f918a4b4b.png)
![gravity3d](https://cloud.githubusercontent.com/assets/3610253/14281862/d92a330a-fb3b-11e5-9b12-e8b5f1f4a0b1.png)

**Comment by sebcrozet** [comment]
Well I am not sure why the current `ConTextSetting` makes it crash on OSX while the default one doesn't.

In fact my suggestion to use the default settings instead comes from reading and mimicking the sfml examples that worked on your system. Besides loosing anti-aliasing I don't know how it will affect the display.

**Comment by sebcrozet** [comment]
Fixed by #58 . Closing.
